nixos/modules/config/nsswitch.nix

```diff
@@ -95,7 +95,9 @@ with lib;
   config = {
     assertions = [
       {
-        # generic catch if the NixOS module adding to nssModules does not prevent it with specific message.
+        # Prevent users from disabling nscd, with nssModules being set.
+        # If disabling nscd is really necessary, it's still possible to opt out
+        # by forcing config.system.nssModules to [].
         assertion = config.system.nssModules.path != "" -> config.services.nscd.enable;
         message = "Loading NSS modules from path ${config.system.nssModules.path} requires nscd being enabled.";
```

Comment:
IMHO that message should also tell you which option triggers this not just the value of it. Mentioning both option names (systemd.nssModules.path and services.nscd.enable) is probably desireable here. You shouldn't need a nixpkgs checkout to figure out what the condition to the message is. Often the code is much cleaner than the messages.

Reply:
Updated.

Reply:
The later version didn't interpolate `config.system.nssModules.path` anymore, so it was harder to figure out what nss modules were configured. I'll change this to the following:

> "Loading NSS modules from system.nssModules.path (${config.system.nssModules.path}) requires services.nscd.enable being set to true."